docs/syntax.md

```diff
@@ -1,22 +1,56 @@
 
-**Kubernetes related steps**:
-1. 	`<GK> a Kubernetes cluster` [AKubernetesCluster](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.AKubernetesCluster)
-2.	`<GK> [I] <create|submit|delete|update> [the] resource <filename>.yaml` [ResourceOperation](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.ResourceOperation)
-3.	`<GK> [I] <create|submit|delete|update> [the] resource <filename>.yaml in [the] <namespacename> namespace` [ResourceOperationInNamespace](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.ResourceOperationInNamespace)
-4.	`<GK> [I] <create|submit|delete|update> [the] resources in <filename>.yaml` [MultiResourceOperation](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.MultiResourceOperation)
-5.	`<GK> [I] <create|submit|delete|update> [the] resources in <filename>.yaml in [the] <namespacename> namespace` [MultiResourceOperationInNamespace](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.MultiResourceOperationInNamespace)
-6.	`<GK> [the] resource <filename> should be <created|deleted>` [ResourceShouldBe](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.ResourceShouldBe)
-7.	`<GK> [the] resource <filename> [should] converge(d) to selector <complete key>=<value>` [ResourceShouldConvergeToSelector](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.ResourceShouldConvergeToSelector)
-8.	`<GK> [the] resource <filename> condition <condition type> should be (true|false)` [ResourceConditionShouldBe](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.ResourceConditionShouldBe)
-9.	`<GK> [I] update [a] resource <filename> with <complete key> set to <value>` [UpdateResourceWithField](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.UpdateResourceWithField)
-10.	`<GK>  <number of> nodes with selector <complete key>=<value> should be (found|ready)` [NodesWithSelectorShouldBe](https://godoc.org/github.com/keikoproj/kubedog/pkg/kubernetes#Client.NodesWithSelectorShouldBe)
-11. `<GK>  [the] <deployment|hpa|horizontalpodautoscaler|service|pdb|poddisruptionbudget> <resource_name> is in namespace <namespace-name>` [ResourceInNamespace](https://pkg.go.dev/github.com/keikoproj/kubedog/pkg/kubernetes#Client.ResourceInNamespace)
-12. `<GK>  [I] scale [the] deployment <filename> in namespace <namespace-name> to <replica(s)>` [ScaleDeployment](https://pkg.go.dev/github.com/keikoproj/kubedog/pkg/kubernetes#Client.ScaleDeployment)
+## Generic steps
+- `<GK> [I] wait for <digits> (minutes|seconds)` common.WaitFor
+- `<GK> the <non-whitespace-characters> command is available` common.CommandExists
+- `<GK> I run the <non-whitespace-characters> command with the <any-characters-except-(")> args the command (fails|succeeds)` common.RunCommand
```

Comment:
This is my mistake, but can we add `and` like below:

```suggestion
- `<GK> I run the <non-whitespace-characters> command with the <any-characters-except-(")> args and the command (fails|succeeds)` common.RunCommand
```

Reply:
This is autogenerated, the change needs to be made in the `kubedog.go` file

Reply:
Right, would you mind changing it there so we don't need another pr?

Reply:
Of course, I did already.